### ISSUE TITLE
Missing space between two links

### ISSUE THREAD

**Issue by arjan** [pragdave/earmark#334]
When creating two adjacent links, it seems a space is missing in the HTML rendering:
```
iex(30)> Earmark.as_html "[link one](aaa) [link two](bbb)"                 
{:ok, "<p><a href=\"aaa\">link one</a><a href=\"bbb\">link two</a></p>\n", []}
```
I would have expected a space between the two `<a>` elements. I think adding this space is according to spec, as the [Commonmark](https://spec.commonmark.org/dingus/) interactive demo does add one.

**Comment by RobertDober** [comment]
thank you for reporting this, but is this space influencing rendering?

**Comment by RobertDober** [comment]
Furthermore this bug will be shadowed by a more profound bug in the 1.5 transformer which will add ws where it should not. 
So let us fix this here too, Really Great Catch @arjan

**Comment by whilefalse** [comment]
Hi, I've noticed another instance of this, I'm guessing it's related. If I have some bold text followed by a space and then a link, I don't get the expected space:

```elixir
iex> Earmark.as_html("**hi** [link](http://example.com)")
{:ok, "<p><strong>hi</strong><a href=\"http://example.com\">link</a></p>\n", []}
```

But if I add an extra character in between the asterix's and the link, I get the space as expected.
```elixir
iex> Earmark.as_html("**hi**s [link](http://example.com)")
{:ok, "<p><strong>hi</strong>s <a href=\"http://example.com\">link</a></p>\n",[]}
```

**Comment by RobertDober** [comment]
The space is missing in the AST and both examples are the same error indeed, thank you both:

Therefor this will be fixed in https://github.com/RobertDober/earmark_parser/issues/10

**Comment by RobertDober** [comment]
fixed with EarmarkParser 1.4.10
